astropy/convolution/convolve.py

```diff
@@ -14,8 +18,44 @@
 from ..nddata import support_nddata
 from ..modeling.core import _make_arithmetic_operator, BINARY_OPERATORS
 from ..modeling.core import _CompoundModelMeta
-
-
+from matplotlib.pyplot import fill_between
+from .utils import KernelSizeError, has_even_axis, raise_even_kernel_exception
+from ..openmp_enabled import is_openmp_enabled
+
+faulthandler.enable()
```

Comment:
I guess `faulthandler` was useful for development but could be removed now ?
Also the matplotlib import above seems unused (otherwise it should be made optional).
And unless I missed something, I don't see the source code for `openmp_enabled`.

Reply:
>I guess faulthandler was useful for development but could be removed now ?

I left the `faulthandler` in to deal with any unforeseen runtime issues, better the user gets an error message than it just quickly and silently returning. I have no strong objection to removing it though if that is the "Python" thing to do.

>Also the matplotlib import above seems unused (otherwise it should be made optional).

Good catch, must have been a copy-and-paste fat-finger.

>And unless I missed something, I don't see the source code for openmp_enabled.

This is autogenerated at build _c.f._  https://github.com/astropy/astropy-helpers/pull/382 [src code here](https://github.com/jamienoss/astropy-helpers/blob/0eaacc90ecd47245050369d7b89a149139368538/astropy_helpers/openmp_helpers.py#L156-L187)

Reply:
faulthandler can be activated at startup time, and I was not sure if activating it by default has some drawback. It seems not actually, so maybe it's fine to keep it.

> This is autogenerated at build c.f. astropy/astropy-helpers#382 src code here

Ok, thanks! This is related to another question (that should maybe go in the astropy-helpers discussion): I was also working on building binary wheels for my package with the C extension with OpenMP, and there will be the same issue for Astropy wheels: if the wheels are built with OpenMP support, this does not necessary mean that the system has OpenMP support. It should be fine on Linux, and maybe the OpenMP support in Clang is old enough now to activate OpenMP by default ?

Reply:
>  It should be fine on Linux, and maybe the OpenMP support in Clang is old enough now to activate OpenMP by default

I think llvm/clang version 3+ supports OpenMP by default, it just doesn't ship libomp with it like gcc does. You have to install it separately, which is why it requires more work to compile & link.